src/bft.py

```diff
@@ -12,4 +12,20 @@ def bf_order(t: T | None) -> Iterable[int]:
     >>> list(bf_order(tree))
     [2, 1, 4, 3, 5]
     """
-    return  # FIXME
+    order = []
+    q = []
+
+    while(True):
```

Comment:
You usually don't have parentheses in

```python
while <something>:
    ...
```

loops in Python.

I really like the way you loop, though, but since you have to remember to pop in each branch (until you break), you might want to make it part of the loop condition so you never forget.

I haven't tested this, but I think this should work:

```python
q = deque([t] if t is not None else []) # a deque because the `pop(0)` stuff is a *bad* idea
while q
    t = q.pop()
    order.append(t.val)
    if t.left:
      q.append(t.left)
    if t.right:
      q.append(t.right)
```

We will iterate as long as the queue is non-empty, we always pop `t`, and since we never append `None` we don't need to worry about it.

(If we do append `None`, which your code does) then we can add one construction to it:

```python
q = deque([t] if t is not None else [])
while q
    if (t := q.pop()) is not None:
       order.append(t.val)
       q.append(t.left)
       q.append(t.right)
```

We use the walrus operation (`:=`) to get `t` out of the queue and at the same time check if it is `None`. Only if it isn't `None` do we process it.